### PR TITLE
Support KIND_CLUSTER_NAME.

### PR DESCRIPTION
**Pull request by amwat**
fixes https://github.com/kubernetes-sigs/kind/issues/1094

/cc @BenTheElder

### CODE REVIEW

File: pkg/cmd/kind/create/cluster/createcluster.go [hunk 1]
```diff
@@ -51,6 +52,9 @@ func NewCommand(logger log.Logger, streams cmd.IOStreams) *cobra.Command {
 		Short: "Creates a local Kubernetes cluster",
 		Long:  "Creates a local Kubernetes cluster using Docker container 'nodes'",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if !cmd.Flags().Changed("name") {
```
Comment (BenTheElder) [inline pkg/cmd/kind/create/cluster/createcluster.go:55]: we could dry this out by just passing the `cmd.Flags()` (flagset pointer) to `OverrideDefaultName` and then internally check `flags.Changed("name")` then call `flags.Set("name", value)`

File: pkg/internal/cli/override.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"os"
+
+	"sigs.k8s.io/kind/pkg/log"
+)
+
+func OverrideDefaultName(logger log.Logger, defaultName *string) {
+	if name := os.Getenv("KIND_CLUSTER_NAME"); name != "" {
+		*defaultName = name
+		logger.V(1).Infof("using %q due to KIND_CLUSTER_NAME", name)
```
Comment (BenTheElder) [inline pkg/internal/cli/override.go:12]: using what? this message seems a little obtuse.

File: pkg/cmd/kind/create/cluster/createcluster.go (hunk 1 above)
Comment (amwat) [reply pkg/cmd/kind/create/cluster/createcluster.go]: Done.
I initially had that, but figured that checking the specific `name` flag logic should probably live near the command than in the method that just checks the env.
/shrug

File: pkg/internal/cli/override.go (hunk 2 above)
Comment (amwat) [reply pkg/internal/cli/override.go]: Clarified.

File: pkg/internal/cli/override.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+	"github.com/spf13/pflag"
+	"sigs.k8s.io/kind/pkg/log"
+)
+
+func OverrideDefaultName(logger log.Logger, fs *pflag.FlagSet) {
+	if !fs.Changed("name") {
+		if name := os.Getenv("KIND_CLUSTER_NAME"); name != "" {
+			fs.Set("name", name)
+			logger.V(1).Infof("using default cluster name as %q due to KIND_CLUSTER_NAME", name)
```
Comment (BenTheElder) [inline pkg/internal/cli/override.go:14]: Last nit: it's not the default right?
using cluster name %q from KIND_CLUSTER_NAME

Or just drop this all together. I don't think most apps log this sort of thing.

Comment (amwat) [reply pkg/internal/cli/override.go]: done

**Comment by BenTheElder** [comment]
/lgtm
/approve
Maybe write elaborate unit tests in a follow up ;-)
Thanks!

**Pull request merged**